src/components/InputField/InputField.test.tsx

```diff
@@ -33,4 +33,54 @@ describe('<InputField />', () => {
 
     expect(onChange).toHaveBeenCalledTimes(testText.length);
   });
+
+  it('will not fire when maxLength is reached', async () => {
+    const user = userEvent.setup();
+    const onChange = jest.fn();
+    const testText = 'typing';
+
+    render(
+      <InputField
+        aria-label="label"
+        data-testid="test-input"
+        defaultValue={testText}
+        maxLength={6}
+        onChange={onChange}
+      />,
+    );
+    const input = screen.getByTestId('test-input');
+
+    input.focus();
+
+    await act(async () => {
+      await user.keyboard(testText);
+    });
+
+    expect(onChange).toHaveBeenCalledTimes(0);
+  });
+
+  it('will fire when recommendedMaxLength is reached', async () => {
+    const user = userEvent.setup();
+    const onChange = jest.fn();
+    const testText = 'typing';
+
+    render(
+      <InputField
+        aria-label="label"
+        data-testid="test-input"
+        defaultValue={testText}
+        onChange={onChange}
+        recommendedMaxLength={6}
+      />,
+    );
+    const input = screen.getByTestId('test-input');
+
+    input.focus();
+
+    await act(async () => {
+      await user.keyboard(testText);
+    });
```

Comment:
(Asked this on another PR, but don't remember the reponse)

What happens if we remove the `act`? IIRC, `user-event` is supposed to handle that for us.

Reply:
Yah, something throws warnings in the test runner when `act` is missing, tho the test itself seems to past. 🤔 Mostly adding it to keep the output clean.